lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -212,3 +212,23 @@ end
 -- @hook
 -- @realm client
 function CLGAMEMODEMENU:Initialize() end
+
+---
+-- Populates the nav panel contents. The default is what is typically desired,
+-- however this extension point enables certain menus (like the Roles menu)
+-- to be populated in a more fitting way.
+-- @note This function can be overridden to create a custom nav panel.
+-- @param Panel navPanel The nav panel container
+-- @param Panel contentPanel The content panel container
+-- @hook
+-- @realm client
+function CLGAMEMODEMENU:FillNavPanel(navPanel, contentPanel, padding)
+    -- MAKE SEPARATE SUBMENULIST ON THE NAVAREA WITH A CONTENT AREA
+    local submenuList = vgui.Create("DSubmenuListTTT2", navPanel)
+    submenuList:Dock(FILL)
+    submenuList:SetPadding(padding)
+    submenuList:SetBasemenuClass(self, contentPanel)
+
+    -- REFRESH SIZE OF SUBMENULIST FOR CORRECT SUBMENU DEPENDENT SIZE
+    submenuList:InvalidateLayout(true)
+end
```

Comment:
Is this still in use? I think this is a leftover from a merge conflict, isn't it? See this comment: https://github.com/TTT-2/TTT2/pull/1702#discussion_r1915373028

I can't find the place where you call this from

Reply:
Argh, this is why I dislike squashing. Fixing.

Reply:
Understandable. But without squashing we'd need all contributors to have clean commits, because we don't want stuff like this in the history:

![image](https://github.com/user-attachments/assets/e1c657e1-2985-43f8-a23b-5196871873f7)
(No worries, my commits look the same)

Histalek for example follows clean commit strategies, but this is (in my opinion) too advanced for the average contributor. Moreover, this project already has more than 2000 commits with squashed PRs. Without it, we would be way north of 10k. Which isn't a bad thing per se, but probably makes tracking down problems in the history even harder.

Reply:
I personally dislike rewriting git history in general. and would rather a messy but accurate commit history over a clean but less-than-helpful history.

Reply:
Take a look at our first 500 commits and then tell me if you stand by your opinion :D